src/test/java/GameConsoleTest.java

```diff
@@ -0,0 +1,30 @@
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class GameConsoleTest {
+
+    @Test
+    public void readMethodReadsAString() {
+        InputStream in = new ByteArrayInputStream("Hello".getBytes());
+        GameConsole gameConsole = new GameConsole(in, null);
+
+        assertEquals("Hello", gameConsole.read());
+    }
+
+    @Test
+    public void writeMethodWritesAString() {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        PrintStream out = new PrintStream(outputStream);
+
+        GameConsole echoConsole = new GameConsole(null, out);
```

Comment:
You were probably thinking about your echo server here. I'd prefer being consistent with the name of the variable for your instance of the class under test. This is really a writingConsole, echo would repeat it after being told it.